pkg/plugin/server_mux.go

```diff
@@ -0,0 +1,65 @@
+package plugin
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// ServerInitializer is a function that initializes and returns a new instance of one of Ark's plugin interfaces
+// (ObjectStore, BlockStore, BackupItemAction, RestoreItemAction).
+type ServerInitializer func(logger logrus.FieldLogger) (interface{}, error)
+
+// serverMux manages multiple implementations of a single plugin kind, such as pod and pvc BackupItemActions.
+type serverMux struct {
+	kind         PluginKind
+	initializers map[string]ServerInitializer
+	instances    map[string]interface{}
```

Comment:
WDYT about using `handler` in place of `instance`?

Reply:
+1